akka-javasdk/src/main/scala/akka/javasdk/impl/eventsourcedentity/EventSourcedEntityImpl.scala

```diff
@@ -0,0 +1,295 @@
+
+  private val componentDescriptor = ComponentDescriptor.descriptorFor(componentClass, messageCodec)
+
+  // FIXME remove EventSourcedEntityRouter altogether, and only keep stateless ReflectiveEventSourcedEntityRouter
+  private def createRouter(context: EventSourcedEntityContext)
+      : ReflectiveEventSourcedEntityRouter[AnyRef, AnyRef, EventSourcedEntity[AnyRef, AnyRef]] =
+    new ReflectiveEventSourcedEntityRouter[S, E, ES](
+      factory(context),
+      componentDescriptor.commandHandlers,
+      messageCodec)
+      .asInstanceOf[ReflectiveEventSourcedEntityRouter[AnyRef, AnyRef, EventSourcedEntity[AnyRef, AnyRef]]]
+
+  override def emptyState(entityId: String): SpiEventSourcedEntity.State = {
+    // FIXME rather messy with the contexts here
+    val cmdContext =
+      new CommandContextImpl(entityId, 0L, "", 0, false, MetadataImpl.of(Nil), None, tracerFactory)
+    val context = new EventSourcedEntityContextImpl(entityId)
+    val router = createRouter(context)
+    router.entity._internalSetCommandContext(Optional.of(cmdContext))
+    try {
+      router.entity.emptyState()
+    } finally {
+      router.entity._internalSetCommandContext(Optional.empty())
+    }
+  }
+
+  override def handleCommand(
+      state: SpiEventSourcedEntity.State,
+      command: SpiEntity.Command): Future[SpiEventSourcedEntity.Effect] = {
+    val entityId = command.entityId
+
+    val span: Option[Span] = None // FIXME traceInstrumentation.buildSpan(service, command)
+    span.foreach(s => MDC.put(Telemetry.TRACE_ID, s.getSpanContext.getTraceId))
+    val cmd =
+      messageCodec.decodeMessage(
+        command.payload.getOrElse(
+          // FIXME smuggling 0 arity method called from component client through here
+          ScalaPbAny.defaultInstance.withTypeUrl(AnySupport.JsonTypeUrlPrefix).withValue(ByteString.empty())))
+    val metadata: Metadata =
+      MetadataImpl.of(Nil) // FIXME MetadataImpl.of(command.metadata.map(_.entries.toVector).getOrElse(Nil))
+    val cmdContext =
+      new CommandContextImpl(
+        entityId,
+        command.sequenceNumber,
+        command.name,
+        0,
+        command.isDeleted,
+        metadata,
+        span,
+        tracerFactory)
+
+    val context = new EventSourcedEntityContextImpl(entityId)
+    val router = createRouter(context)
+    router.entity._internalSetCommandContext(Optional.of(cmdContext))
+    try {
+      router.entity._internalSetCurrentState(state)
+      val commandEffect = router
+        .handleCommand(command.name, state, cmd, cmdContext)
+        .asInstanceOf[EventSourcedEntityEffectImpl[AnyRef, E]] // FIXME improve?
+
+      def replyOrError(updatedState: SpiEventSourcedEntity.State): (Option[ScalaPbAny], Option[SpiEntity.Error]) = {
+        commandEffect.secondaryEffect(updatedState) match {
+          case ErrorReplyImpl(description, status) =>
+            val errorCode = status.map(_.value).getOrElse(Status.Code.UNKNOWN.value)
```

Comment:
This is legacy. 

The Effect API doesn't allow the usage of an error code anymore, but the protocol is still requiring it. 

I'm sure that this will always be a `None` and now falling back to `Status.Code.UNKNOWN`.

We should remove this completely.

Reply:
I'll have to test these error scenarios, because something wasn't working without the error code